refactor: fix mypy errors in package_list_parser.py

We're in the middle of trying to improve the type hints we use in cve-bin-tool.  I previously filed some issues as found with pyright, this particular set of issues is done using mypy.  We're new to type checking, so it's entirely possible that the types currently in the file are incorrect or incomplete. To test the file yourself, you'll need to install mypy (e.g. `pip install mypy`) and then run it with the filename you want to test.   You may need to also run `mypy --install-types` to get it to work on some files.

Note that I'm filing issues for each file individually to help folk who are looking for fun stuff to do as part of hacktoberfest so you can claim one and not interfere with anyone else working on another file.  If you've already got your 4 commits for hacktoberfest, please leave these issues for someone who's still working on theirs!

Here's a list of issues found:

```
cve_bin_tool/package_list_parser.py:46: error: Need type annotation for "parsed_data_without_vendor"
cve_bin_tool/package_list_parser.py:47: error: Need type annotation for "parsed_data_with_vendor"
cve_bin_tool/package_list_parser.py:48: error: Need type annotation for "package_names_with_vendor" (hint: "package_names_with_vendor: List[<type>] = ...")
cve_bin_tool/package_list_parser.py:49: error: Need type annotation for "package_names_without_vendor" (hint: "package_names_without_vendor: List[<type>] = ...")
```

Helpful resources:
* [Python typing documentation](https://docs.python.org/3/library/typing.html)
* [mypy cheat sheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)
* If you can't find the answer in the docs, googling the error messages above will often help!
* [cve-bin-tool's contributor docs](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md) 
* If you've contributed to open source but not this project, you might just want our [checklist for a great pull request](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#checklist-for-a-great-pull-request)

Tiny tips for new contributors:
* cve-bin-tool uses https://www.conventionalcommits.org/ style for commit messages, and we have a test that checks the title of your PR.  A good potential title for this one is in the title of this issue.
* You can make an issue auto close by including a comment "fixes #ISSUENUMBER" in your PR comments where ISSUENUMBER is the actual number of the issue.

--- a/cve_bin_tool/package_list_parser.py
+++ b/cve_bin_tool/package_list_parser.py
@@ -7,6 +7,7 @@ from collections import defaultdict
 from logging import Logger
 from pathlib import Path
 from subprocess import PIPE, run
+from typing import Any, Dict, List
 
 import distro
 
@@ -43,10 +44,10 @@ class PackageListParser:
             self.logger = logger
 
         self.error_mode = error_mode
-        self.parsed_data_without_vendor: dict = defaultdict(dict)
-        self.parsed_data_with_vendor: dict = defaultdict(dict)
-        self.package_names_with_vendor: list = []
-        self.package_names_without_vendor: list = []
+        self.parsed_data_without_vendor: Dict[Any, Any] = defaultdict(dict)
+        self.parsed_data_with_vendor: Dict[Any, Any] = defaultdict(dict)
+        self.package_names_with_vendor: List[Any] = []
+        self.package_names_without_vendor: List[Any] = []
 
     def parse_list(self):
         input_file = self.input_file

--- a/cve_bin_tool/package_list_parser.py
+++ b/cve_bin_tool/package_list_parser.py
@@ -43,10 +43,10 @@ class PackageListParser:
             self.logger = logger
 
         self.error_mode = error_mode
-        self.parsed_data_without_vendor = defaultdict(dict)
-        self.parsed_data_with_vendor = defaultdict(dict)
-        self.package_names_with_vendor = []
-        self.package_names_without_vendor = []
+        self.parsed_data_without_vendor: dict = defaultdict(dict)
+        self.parsed_data_with_vendor: dict = defaultdict(dict)
+        self.package_names_with_vendor: list = []
+        self.package_names_without_vendor: list = []
 
     def parse_list(self):
         input_file = self.input_file

